packages/commerce-sdk-react/src/hooks/ShopperBaskets/helpers.test.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React from 'react'
+import {useShopperBasketsMutationHelper} from './helpers'
+import {useCustomerBaskets} from '../ShopperCustomers'
+import {renderWithProviders} from '../../test-utils'
+import {screen, waitFor} from '@testing-library/react'
+import jwt from 'jsonwebtoken'
+
+const basketId = '10cf6aa40edba4fcfcc6915594'
+//  order is important in this mock since it will be called in this order from all the tests
+const mockAsyncMutate = jest
+    .fn()
+    .mockImplementationOnce(() => ({
+        productItems: [{id: 'product-id-111', quantity: 20}],
+        basketId
+    }))
+    .mockImplementationOnce(() => ({
+        productItems: [],
+        basketId
+    }))
+    .mockImplementationOnce(() => ({
+        productItems: [{id: 'product-id', quantity: 2}],
+        basketId
+    }))
```

Comment:
Overall these tests are much easier to read 👍 

I have a question about this code tho. Are we able to mock the implementation for this method in the test itself? The reason I'm acting is because the tests are now order dependant on how we mocked these implementations, which would probably lead to confusion if another developer was to come in to fix or add something in the future. 

If this isn't a possibility I would make sure that you are loud about comment on the importance of the order of these tests and not to change them.

Reply:
Good point. I've moved them into each test as suggested